server.go

```diff
@@ -197,11 +217,13 @@ func (s *Server) Serve(lis net.Listener) error {
 	for {
 		c, err := lis.Accept()
 		if err != nil {
+			s.printf("done serving; Accept = %v", err)
 			return err
 		}
 		if creds, ok := s.opts.creds.(credentials.TransportAuthenticator); ok {
 			c, err = creds.ServerHandshake(c)
```

Comment:
This seems on an out-of-date version. See https://github.com/grpc/grpc-go/blob/master/server.go#L204. You probably need to sync your client.